flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -270,16 +269,9 @@ public TaskManagerLocation getCurrentAssignedResourceLocation() {
         return currentExecution.getAssignedResourceLocation();
     }
 
-    @Nullable
     @Override
-    public ArchivedExecution getPriorExecutionAttempt(int attemptNumber) {
-        synchronized (priorExecutions) {
-            if (attemptNumber >= 0 && attemptNumber < priorExecutions.size()) {
-                return priorExecutions.get(attemptNumber);
-            } else {
-                throw new IllegalArgumentException("attempt does not exist");
-            }
-        }
+    public ExecutionHistory getExecutionHistory() {
+        return executionHistory;
```

Comment:
I'm not quite sure what the `synchronized` is for,  is it safe to delete it?

Reply:
ExecutionGraph used to be possible to be concurrently modified&accessed. Now with the new JM thread model(single-threaded of JM status access&modification), concurrency would not happen anymore.